lib/errors.ts

```diff
@@ -0,0 +1,73 @@
+type ErrorOpts = {
+  message?: string;
+  status?: number;
+  type?: ErrorType;
+};
+
+export enum ErrorType {
```

Comment:
Do we really need this `ErrorType` enum?
I think we might just use `instanceOf` or do I miss sth.?

```typescript
type ErrorOpts = {
    message?: string;
    status?: number;
};

class KontistSDKError extends Error {
    public status?: number;

    constructor(opts: ErrorOpts = {}) {
        super(opts.message);
        this.status = opts.status;
    }
}

class ChallengeExpiredError extends KontistSDKError {
    constructor(opts: ErrorOpts = {}) {
        super(opts);
        Object.setPrototypeOf(this, ChallengeExpiredError.prototype);
    }
}

const e = new ChallengeExpiredError({ status: 42 });

console.log(e instanceof ChallengeExpiredError ?
    'is instanceof ChallengeExpiredError' : 'not instanceof ChallengeExpiredError')

throw e;
```

Outputs:
```
is instanceof ChallengeExpiredError
/Users/johannes/tmp_error/index.js:37
throw e;
^

ChallengeExpiredError
    at ChallengeExpiredError.KontistSDKError [as constructor] (/Users/johannes/tmp_error/index.js:19:28)
    at new ChallengeExpiredError (/Users/johannes/tmp_error/index.js:29:28)
    at Object.<anonymous> (/Users/johannes/tmp_error/index.js:35:9)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11 {
  status: 42
```

Reply:
No you're right, we don't really need it, I just wasn't sure what would be the preferred way of handling these errors so I added a bunch of different possible ways. Let's stick to `instanceof` and remove this `type` property.

a9c9f2419fc2d39b751a7219658eeffb86920359